opencog/spacetime/octomap/OctoMapValue.cc

```diff
@@ -0,0 +1,69 @@
+#include "OctoMapValue.h"
+
+#include <opencog/spacetime/octomap/atom_types.h>
+#include <opencog/util/exceptions.h>
+#include <opencog/atoms/base/ClassServer.h>
+#include <opencog/atoms/value/ValueFactory.h>
+
+using namespace opencog;
+
+
+void OctoValue::update()
+{
+    //point3d p = (_om->get_locations_of_atom(_octo_atom)).front();
+    //_value = {p.x(), p.y(), p.z()};
+    _om->insert_atom(point3d(_value[0], _value[1], _value[2]), _item);
```

Comment:
OK, well, this is clearly wrong -- it needs to be the other way around: when `update` is called, it should ask the octomap what the correct values should be. That is, it should look something like this:
```
void OctoValue::update() {
     _value = _om->get_current_location(_item);
}
```